packages/core/src/types/redocly-yaml.ts

```diff
@@ -0,0 +1,128 @@
+import { NodeType } from '.';
+
+const ConfigRoot: NodeType = {
+  properties: {
+    apiDefinitions: { type: 'object', additionalProperties: { type: 'string' } },
+    lint: 'Lint',
+    referenceDocs: 'ReferenceDocs',
+  },
+  required: ['apiDefinitions'],
+};
+
+const Lint: NodeType = {
+  properties: {
+    plugins: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+    extends: {
+      type: 'array',
+      items: {
+        type: 'string',
+      },
+    },
+    doNotResolveExamples: { type: 'boolean' },
+    rules: { type: 'object'},
+    oas2Rules: { type: 'object'},
+    oas3_0Rules: { type: 'object'},
+    oas3_1Rules: { type: 'object'},
+    preprocessors: { type: 'object'},
+    oas2Preprocessors: { type: 'object'},
+    oas3_0Preprocessors: { type: 'object'},
+    oas3_1Preprocessors: { type: 'object'},
+    decorators: { type: 'object'},
+    oas2Decorators: { type: 'object'},
+    oas3_0Decorators: { type: 'object'},
+    oas3_1Decorators: { type: 'object'},
+    resolve: {
```

Comment:
Don't do deep types (deeper than 2 for arrays), it's not supported by the spec rule I believe (not 100% sure). Instead have named types.